### PR TITLE
IGNORE: trying to get tekton context to fail

### PR DESCRIPTION
**Pull request by abayer**
Specifically to demonstrate #6812, which has been masked by
reproduce #6812 enough to fix it and then go deal with #6811 in the future.

### CODE REVIEW

File: pkg/cmd/step/helm/step_helm_apply.go [hunk 1]
```diff
@@ -406,18 +406,17 @@ func (o *StepHelmApplyOptions) getRequirements() (*config.RequirementsConfig, st
...
 	}
 
-	// TODO To stay backwards compatible with code after fixing #6653, we need to create a dummy RequirementsConfig here
-	// TODO This is a special case for static master and needs to be removed (HF)
-	if !teamSettings.IsJenkinsXPipelines() {
+	requirements, err = config.GetRequirementsConfigFromTeamSettings(teamSettings)
+	if err != nil {
+		return nil, "", errors.Wrap(err, "getting the requirements from team settings")
+	}
+	// TODO: Workaround for non-boot clusters. Remove when we get rid of jx install. (APB)
```
Comment (hferentschik) [inline pkg/cmd/step/helm/step_helm_apply.go]: +1, that's what I had in mind as well. Is this not working?

File: pkg/jenkinsfile/pipeline.go [hunk 2]
```diff
@@ -933,6 +933,10 @@ func (c *PipelineConfig) CreatePipelineForBuildPack(args CreatePipelineArguments
 	// If agent.container is specified, use that for default container configuration for step images.
 	containerName := c.Agent.GetImage()
 	if containerName != "" {
+		gcrRemoved := strings.TrimPrefix(containerName, "gcr.io/jenkinsxio/builder-")
```
Comment (hferentschik) [inline pkg/jenkinsfile/pipeline.go]: Is this related?

Comment (abayer) [reply pkg/jenkinsfile/pipeline.go]: See #6811 - the first commit in this PR is me finding a hackish way to actually test the PR image in the `charts` build pack. Just a hack, not to be merged - we need a real solution for #6811 down the road.

File: pkg/cmd/step/helm/step_helm_apply.go (hunk 1 above)
Comment (abayer) [reply pkg/cmd/step/helm/step_helm_apply.go]: It is. =)

Comment (abayer) [reply pkg/cmd/step/helm/step_helm_apply.go]: Rather, it _is_ working.
